internal/service/ipam/model_ipamsvc_ha_group_host.go

```diff
@@ -0,0 +1,121 @@
+package ipam
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	schema "github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+
+	"github.com/infobloxopen/bloxone-go-client/ipam"
+
+	"github.com/infobloxopen/terraform-provider-bloxone/internal/flex"
+)
+
+type IpamsvcHAGroupHostModel struct {
+	Address    types.String `tfsdk:"address"`
+	Heartbeats types.List   `tfsdk:"heartbeats"`
+	Host       types.String `tfsdk:"host"`
+	Port       types.Int64  `tfsdk:"port"`
+	Role       types.String `tfsdk:"role"`
+	State      types.String `tfsdk:"state"`
+}
+
+var IpamsvcHAGroupHostAttrTypes = map[string]attr.Type{
+	"address":    types.StringType,
+	"heartbeats": types.ListType{ElemType: types.ObjectType{AttrTypes: IpamsvcHAGroupHeartbeatsAttrTypes}},
+	"host":       types.StringType,
+	"port":       types.Int64Type,
+	"role":       types.StringType,
+	"state":      types.StringType,
+}
+
+var IpamsvcHAGroupHostResourceSchemaAttributes = map[string]schema.Attribute{
+	"address": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "The address on which this host listens.",
+	},
+	"heartbeats": schema.ListNestedAttribute{
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: IpamsvcHAGroupHeartbeatsResourceSchemaAttributes,
+		},
+		Computed:            true,
+		MarkdownDescription: "Last successful heartbeat received from its peer/s. This field is set when the _collect_stats_ is set to _true_ in the _GET_ _/dhcp/ha_group_ request.",
+	},
+	"host": schema.StringAttribute{
+		Required:            true,
+		MarkdownDescription: "The resource identifier.",
+	},
+	"port": schema.Int64Attribute{
+		Computed:            true,
+		MarkdownDescription: "The HA port.",
+	},
+	"role": schema.StringAttribute{
+		Required: true,
+		Validators: []validator.String{
+			stringvalidator.OneOf("active", "passive"),
+		},
+		MarkdownDescription: "The role of this host in the HA relationship: _active_ or _passive_.",
+	},
+	"state": schema.StringAttribute{
+		Computed:            true,
+		MarkdownDescription: "The state of DHCP on the host. This field is set when the _collect_stats_ is set to _true_ in the _GET_ _/dhcp/ha_group_ request.",
```

Comment:
are we setting `collect_stats` as true anywhere in our requests, or can the user provide it?

Reply:
Something the user provides on the `list` and `get` calls. Let me check if I have to make changes to the generator and go-client repo to bring this field.

Reply:
This is particularly useful for clients who have a large set of HA Groups to build automation to monitor the status/state of HA using Terraform.